base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/editing/blocks/BlocksEditorPanel.java

```diff
@@ -142,15 +163,59 @@ public void mouseDoubleClick(MouseEvent e) {
                 }
             }
         });
+        
+        /*
+         * Allows for a keyboard shortcut (Ctrl+c) to copy block.
+         */
+        table.addKeyListener(new KeyListener() {
+            @Override
+            public void keyPressed(KeyEvent e) {          
+            }
+            
+            @Override
+            public void keyReleased(KeyEvent e) {
+                if (e.character == 0x3) {
+                    copySelected();
+                }
+            }
+        });
 	}
-
+	
+	/**
+     * This method is responsible for setting the config to which the blocks will be added.
+     * 
+     * @param config the configuration the blocks are to be registered with.
+     */
 	public void setConfig(EditableConfiguration config) {	
 		this.config = config;
 		
 		add.setEnabled(true);
 		setBlocks(config);
 	}
-
+	
+	/**
+     * This method is responsible for copying blocks.
+     * 
+     */
+	public void copySelected() {
+	    if (!table.selectedRows().isEmpty()) {
+    	    for (Block block : table.selectedRows()) {
+                EditableBlock added = new EditableBlock(block);
+                added.setName(viewModel.getUniqueName(added.getName()));
+                try {
+                    config.addNewBlock(added);
+                } catch (DuplicateBlockNameException e1) {
+                    // TODO Auto-generated catch block
+                    e1.printStackTrace();
```

Comment:
Remove `TODO` and add more appropriate error handling. At a minimum send it to the `IsisLog`, but really you probably want to bring up an error dialog so that the user notices.